exchange/auction.go

```diff
@@ -15,6 +16,29 @@ import (
 	"github.com/prebid/prebid-server/prebid_cache_client"
 )
 
+type DebugLog struct {
+	EnableDebug bool
+	CacheType   prebid_cache_client.PayloadType
+	Data        DebugData
+	TTL         int64
+	CacheKey    string
+	CacheString string
+}
+
+type DebugData struct {
+	Request  string
+	Headers  string
+	Response string
+}
+
+func (d *DebugLog) BuildDebugLog() {
+	d.Data.Request = fmt.Sprintf("<Request>%s</Request>", d.Data.Request)
```

Comment:
Is it possible for the request, headers, or response string contains invalid xml?

Reply:
Added a regex replacer for `<>` characters in case. This is used before the tags are added, so it should fix formatting issues if those characters are included. Let me know if this is too aggressive of a replacer.

Reply:
Probably want to compile the `matcher` when you build the `DebugLog` object, and store it there, so we don't have to recompile every time this function is called.

Reply:
Added `Matcher` as a new field. Updated the construction to compile the regex.

Reply:
Awesome. The name 'Matcher' seems a bit ambiguous at that level, but other names I would suggest are awful.. like CacheStringNormlizationMatcher. Thoughts @hhhjort?

Reply:
A little bit of a descriptive name would be good, especially in light of my next suggestion.

Reply:
We probably should look into moving the matcher all the way into the `endpointDeps`, as it stands now we will recompile this for every request still, just locked down to only once per request. The other endpoints should be able to ignore the additional `endpointDeps` field, at least until they have a reason to use it.

Reply:
Ah good point! Moved the regexp to be in `endpointDeps` and renamed the field from `Matcher` to just `Regexp`.

Reply:
You could move the compiling into `NewVideoEndpoint()` rather than compiling it router and passing it in. Saves one argument in the function, but that is a minor nit pick at this point.